core/domain/suggestion_services.py

```diff
@@ -542,3 +556,55 @@ def check_can_resubmit_suggestion(suggestion_id, user_id):
     suggestion = get_suggestion_by_id(suggestion_id)
 
     return suggestion.author_id == user_id
+
+
+def check_can_edit_suggestion(user, suggestion_id):    # pylint: disable=too-many-return-statements
+    """Checks whether the user have permission to edit suggestion or not.
+
+    Args:
+        user: UserActionsInfo. Object having user_id, role and actions for
+            given user.
+        suggestion_id: str. Id of a suggestion.
+
+    Returns:
+        bool: Whether the user can resubmit the suggestion.
+    """
+
+    suggestion = get_suggestion_by_id(suggestion_id)
+    user_id = user.user_id
+
+    if suggestion.author_id == user_id:
+        return True
+
+    if suggestion.target_type == 'exploration':
+        exploration_rights = rights_manager.get_exploration_rights(
+            suggestion.target_id)
+        if exploration_rights is None:
+            return False
+
+        if rights_manager.check_can_edit_activity(
+                user, exploration_rights):
+            return True
+
+    elif suggestion.target_type == 'collection':
+        collection_rights = rights_manager.get_collection_rights(
+            suggestion.target_id)
+        if collection_rights is None:
+            return False
+
+        if rights_manager.check_can_edit_activity(
+                user, collection_rights):
+            return True
+
+    elif suggestion.target_type == 'question':
+        question_rights = question_services.get_question_rights(
+            suggestion.target_id, strict=False)
+        if question_rights is None:
+            return False
+
+        if (
+                role_services.ACTION_EDIT_ANY_QUESTION in user.actions or
+                question_rights.is_creator(user.user_id)):
+            return True
+
```

Comment:
Add a final else clause that will throw an error for incorrect target types

Reply:
done
PTAL @nithusha21